diff --git a/linenoise.c b/linenoise.c
index 8c72b9a..9c6d8d1 100644
--- a/linenoise.c
+++ b/linenoise.c
@@ -106,7 +106,8 @@
 #include <sys/types.h>
 #include <sys/ioctl.h>
 #include <unistd.h>
 #include <stdint.h>
+#include <poll.h>
 #include "linenoise.h"
 
 #define LINENOISE_DEFAULT_HISTORY_MAX_LEN 100
@@ -637,4 +638,5 @@ static int getCursorPosition(int ifd, int ofd) {
     /* Read the response: ESC [ rows ; cols R */
     while (i < sizeof(buf)-1) {
+        if (poll(&(struct pollfd){ifd, POLLIN, 0}, 1, 250) <= 0) return -1;
         if (read(ifd,buf+i,1) != 1) break;
         if (buf[i] == 'R') break;
